modules/protocol/resources/META-INF/smithy/smithy4s.meta.smithy

```diff
@@ -163,3 +163,14 @@ structure generateOptics {}
 ///  via extending scala.util.control.NoStackTrace instead of Throwable.
 @trait(selector: "structure :is([trait|error])")
 structure noStackTrace {}
+
+/// Allows users to manually add imports to files of generated shapes.
+/// This would be helpful when some shape needs a specific import in order
+/// to compile. espically in the case you want to compose refinement types
+/// and other validators. `providerImport` should be an import that the
+/// target shape is required to compile.
+@trait(selector: "structure :not([trait|error])") // todo remove trait
```

Comment:
For now, I only limit it to `structure`, where I guess it makes most sense (and make it easier to implement). But maybe I should open it to `union` and `newtype` as well.

Reply:
You can lift the constraint altogether imho

Reply:
do you mean it should just be: `@trait(selector: "*")`?

Reply:
I mean it should just be `@trait`, no selector needed.

Reply:
fixed in https://github.com/disneystreaming/smithy4s/pull/1550/commits/5873cc511a505f720e01c7e9de6b812a03feee9a